src/platform/Darwin/DeviceInstanceInfoProviderImpl.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "DeviceInstanceInfoProviderImpl.h"
+
+#include <platform/CHIPDeviceConfig.h>
+#include <platform/CHIPDeviceError.h>
+#include <platform/Darwin/PosixConfig.h>
+#include <platform/internal/GenericDeviceInstanceInfoProvider.ipp>
+
+namespace chip {
+namespace DeviceLayer {
+
+CHIP_ERROR DeviceInstanceInfoProviderImpl::GetVendorId(uint16_t & vendorId)
+{
+#if CHIP_DISABLE_PLATFORM_KVS
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+#else  // CHIP_DISABLE_PLATFORM_KVS
```

Comment:
Since this is only build if !CHIP_DISABLE_PLATFORM_KVS, we can drop the ifdefs here, I expect....